### ISSUE TITLE
broken display of structured facts

### ISSUE THREAD

**Issue by anarcat** [sodabrew/puppet-dashboard#322]
the following fact:

```
partitions => {"sda1"=>{"size"=>"390626", "filesystem"=>"ext4", "uuid"=>"a0aa7acf-778e-4d2b-9947-e33c0fdec1b4", "mount"=>"/boot"}, "sda2"=>{"size"=>"3906251", "filesystem"=>"ext4", "uuid"=>"34458a9c-edbc-445f-9e54-d1139f4e749c", "mount"=>"/"}, "sda3"=>{"size"=>"3898824219", "filesystem"=>"ext4", "uuid"=>"20b674f5-17ed-4cba-8598-0e8f724cdc0e", "mount"=>"/var/isuma"}, "sda4"=>{"size"=>"3907989", "filesystem"=>"swap", "uuid"=>"ea71f43e-7634-4fe8-8f03-a1ca405df7fd"}}
```

gets turned into this in puppet-dashboard's inventory:

```
sda1size390626filesystemext4uuida0aa7acf-778e-4d2b-9947-e33c0fdec1b4mount/bootsda2size3906251filesystemext4uuid34458a9c-edbc-445f-9e54-d1139f4e749cmount/sda3size3898824219filesystemext4uuid20b674f5-17ed-4cba-8598-0e8f724cdc0emount/var/isumasda4size3907989filesystemswapuuidea71f43e-7634-4fe8-8f03-a1ca405df7fd
```

i believe this may be because the client is running squeeze instead of wheezy, because on a wheezy server, the facts get displayed properly by the dashboard. however, the facter/puppet versions are similar...

working, in debian wheezy:

```
root@koumbit-mp-test:~# facter -v
2.4.3
root@koumbit-mp-test:~# puppet --version
3.7.5
root@koumbit-mp-test:~# ruby --version
ruby 1.9.3p194 (2012-04-20 revision 35410) [x86_64-linux]
```

broken, in debian squeeze:

```
root@host233:~# facter -v
2.4.1
root@host233:~# puppet --version
3.7.4
root@host233:~# ruby --version
ruby 1.8.7 (2010-08-16 patchlevel 302) [i486-linux]
```

ideas?

**Comment by sodabrew** [comment]
That looks like a stringified Hash per Ruby 1.8. My question is where the fact is getting stringified. Is this with the same Dashboard looking at both values?

**Comment by anarcat** [comment]
yes, same dashboard

**Comment by sodabrew** [comment]
Have you checked for known issues with Ruby 1.8.7 and the versions above in the Puppet JIRA?

**Comment by anarcat** [comment]
On 2015-05-07 13:22:58, Aaron Stone wrote:
> Have you checked for known issues with Ruby 1.8.7 and the versions above in the Puppet JIRA?

i haven't, unfortunately.

**Comment by anarcat** [comment]
hmm... i wonder if this isn't due to a different config of the `stringify_facts` in puppet.conf:

https://docs.puppetlabs.com/facter/2.4/fact_overview.html#writing-structured-facts

i tried to put stringify_facts=false in puppet.conf on the client and server side, no luck.

**Comment by bwitt** [comment]
hi @anarcat I believe this is caused by ruby 1.8.7 and updating to a supported ruby would fix this (>= 2.2) so I'm going to close this; if that's not the case and it's still an issue on newer ruby, please reopen.  Thanks!
